datahub-web-react/src/app/search/SearchPage.tsx

```diff
@@ -54,6 +68,45 @@ export const SearchPage = () => {
     const page: number = params.page && Number(params.page as string) > 0 ? Number(params.page as string) : 1;
     const filters: Array<FacetFilterInput> = useFilters(params);
 
+    const allSearchResultsByType = useGetAllEntitySearchResults({
+        query,
+        start: 0,
+        count: RESULTS_PER_GROUP,
+        filters: null,
+    });
+
+    const loading = Object.keys(allSearchResultsByType).some((type) => {
+        return allSearchResultsByType[type].loading;
+    });
+
+    const noResults = Object.keys(allSearchResultsByType).every((type) => {
+        return (
+            !allSearchResultsByType[type].loading &&
+            allSearchResultsByType[type].data?.search?.searchResults.length === 0
+        );
+    });
+
+    const resultCounts: SearchResultCounts = useMemo(() => {
+        if (!loading) {
+            let resultCount = 0;
+            const counts: SearchResultCounts = {};
+            Object.keys(allSearchResultsByType).forEach((key) => {
+                if (!allSearchResultsByType[key].loading) {
+                    resultCount += allSearchResultsByType[key].data?.search?.total;
+                    counts[key as EntityType] = allSearchResultsByType[key].data?.search?.total || 0;
+                }
+            });
+
+            analytics.event({
+                type: EventType.SearchResultsViewEvent,
+                query,
+                total: resultCount,
+            });
```

Comment:
please move the analytics event into its own useEffect- it is a bit confusing that this is a side effect of the use memo